components/nodemanager-service/nodemanager.go

```diff
@@ -133,3 +262,33 @@ func newGRPCServer(db *pgdb.DB, connFactory *secureconn.Factory, config *config.
 
 	return s
 }
+
+func createOrUpdateWorkflowSchedule(cerealManager *cereal.Manager, scheduleName string, jobName string, rule *rrule.RRule) error {
+	err := cerealManager.CreateWorkflowSchedule(context.Background(), scheduleName, jobName, nil, true, rule)
+
+	if err == nil {
+		return nil
+	}
+
+	if err == cereal.ErrWorkflowScheduleExists {
+		log.Infof("nodemanager workflow schedule %s already exists, not creating", scheduleName)
```

Comment:
Would it make sense to add a function to cereal itself to update a workflow schedule if the recurrence rule is updated? This feels like code people could write several times.